tests/test_peak_processing.py

```diff
@@ -87,3 +86,161 @@ def test_sum_waveform(records, peak_left, peak_length):
 
     # Finally check that we also can use a selection of peaks to sum
     strax.sum_waveform(peaks, records, np.ones(n_ch), select_peaks_indices=np.array([0]))
+
+
+@settings(deadline=None)
+@given(several_fake_records,
+       st.integers(min_value=10, max_value=400),
+       st.integers(min_value=1000, max_value=2000),
+       st.integers(min_value=1900, max_value=10000),
+       st.integers(min_value=1000, max_value=int(7_000_000)),
+       )
+@example(
+    records=np.array([(0, 1, 1, 0, 0, 0, 0, 0, 0., 0., 0, [1, 0]),
+                      (1, 1, 1, 1, 0, 0, 0, 0, 0., 0., 0, [1, 0])],
+                     dtype=strax.record_dtype(2)),
+    gap_factor=108,
+    max_duration=1000,
+    right_extension=5000,
+    gap_threshold=18000,
+)
+def test_peak_overflow(records,
+                       gap_factor,
+                       right_extension,
+                       gap_threshold,
+                       max_duration,
+                       ):
+
+    """
+    Test that we handle dt overflows in peaks correctly. To this end, we
+        just create some sets of records and copy that set of records
+        for a few times. That way we may end up with a very long
+        artificial set of hits that can be used in the peak building. By
+        setting the peak finding parameters to very strange conditions
+        we are able to replicate the behaviour where a peak would become
+        so large that it cannot be written out correctly due to integer
+        overflow of the dt field,
+    :param records: records
+    :param gap_factor: to create very extended sets of records, just
+        add a factor that can be used to multiply the time field with,
+        to more quickly arrive to a very long pulse-train
+    :param max_duration: max_duration option for strax.find_peaks
+    :param right_extension: option for strax.find_peaks
+    :param gap_threshold: option for strax.find_peaks
+    :return: None
+    """
+
+    # Set this here, no need to test left and right independently
+    left_extension = 0
+    p = np.zeros(0, dtype=strax.peak_dtype())
+    magic_overflow_time = np.iinfo(p.dtype['dt']).max * p.dtype['data'].shape[0]
+    del p
+
+    # Make a single big peak to contain all the records
+    def retrun_1(x):
+        """
+        Return 1 for all of the input that can be used as a parameter
+            for the splitting in natural breaks
+        :param x: any type of array
+        :return: ones * len(array)
+        """
+        ret = np.ones(len(x))
+        return ret
```

Comment:
Can we use a lambda function instead? This looks a bit odd.

Reply:
lambda is very non pep8 right? What would your main motivation be for making this a lambda?

Reply:
Hmm, I to my understand something like: 
```
function(a, b, c labda x: do_something_simple)
```
would be the default example for a lambda function. Like used in `filter`, `map` etc.   It is the only places where I would use them.

Reply:
@WenzDaniel , I just stumbled across the pep8 guidelines, and I must say, I would agree with their reasoning:
https://www.python.org/dev/peps/pep-0008/